modules/glimpseBidAdapter.js

```diff
@@ -1,230 +1,201 @@
+    const auth = getVaultJwt();
+    const referer = getReferer(bidderRequest);
+    const imp = validBidRequests.map(processBidRequest);
+    const fpd = getFirstPartyData();
 
     const data = {
       auth,
       data: {
         referer,
-        gdprConsent,
-        bidRequests,
-        site: firstPartyData.site,
-        user: firstPartyData.user,
-        bidderCode: spec.code,
-      }
-    }
+        imp,
+        fpd,
+      },
+    };
 
     return {
       method: 'POST',
-      url: ENDPOINT,
+      url,
       data: JSON.stringify(data),
       options: {},
-    }
+    };
   },
 
   /**
-   * Parse response from Glimpse server
-   * @param bidResponse {ServerResponse}
+   * Parse http response
+   * @param response {ServerResponse}
    * @returns {Bid[]}
    */
-  interpretResponse: (bidResponse) => {
-    const isValidResponse = isValidBidResponse(bidResponse)
-
-    if (isValidResponse) {
-      const {auth, data} = bidResponse.body
-      setVaultJwt(auth)
-      return data.bids
+  interpretResponse: (response) => {
+    if (isValidResponse(response)) {
+      const { auth, data } = response.body;
+      setVaultJwt(auth);
+      const bids = data.bids.map(processBidResponse);
+      return bids;
     }
-
-    return []
+    return [];
   },
-}
+};
 
 function setVaultJwt(auth) {
-  storageManager.setDataInLocalStorage(LOCAL_STORAGE_KEY.vault.jwt, auth)
+  storageManager.setDataInLocalStorage(LOCAL_STORAGE_KEY.vault.jwt, auth);
 }
 
 function getVaultJwt() {
-  return storageManager.getDataFromLocalStorage(LOCAL_STORAGE_KEY.vault.jwt) || ''
+  return (
+    storageManager.getDataFromLocalStorage(LOCAL_STORAGE_KEY.vault.jwt) || ''
+  );
 }
 
 function getReferer(bidderRequest) {
-  const hasReferer =
-    hasValue(bidderRequest) &&
-    hasValue(bidderRequest.refererInfo) &&
-    hasStringValue(bidderRequest.refererInfo.referer)
-
-  if (hasReferer) {
-    return bidderRequest.refererInfo.referer
-  }
-
-  return ''
+  return bidderRequest?.refererInfo?.referer || '';
 }
 
-function getGdprConsentChoice(bidderRequest) {
-  const hasGdprConsent =
-    hasValue(bidderRequest) &&
-    hasValue(bidderRequest.gdprConsent)
+function buildQuery(bidderRequest) {
+  let url = appendQueryParam(ENDPOINT, 'ver', '$prebid.version$');
 
-  if (hasGdprConsent) {
-    const gdprConsent = bidderRequest.gdprConsent
-    const hasGdprApplies = hasBooleanValue(gdprConsent.gdprApplies)
+  const timeout = config.getConfig('bidderTimeout');
+  url = appendQueryParam(url, 'tmax', timeout);
 
-    return {
-      consentString: gdprConsent.consentString || '',
-      vendorData: gdprConsent.vendorData || {},
-      gdprApplies: hasGdprApplies ? gdprConsent.gdprApplies : true,
-    }
+  if (gdprApplies(bidderRequest)) {
+    const consentString = bidderRequest.gdprConsent.consentString;
+    url = appendQueryParam(url, 'reg', 'gdpr');
+    url = appendQueryParam(url, 'cs', consentString);
+  } else if (ccpaApplies(bidderRequest)) {
```

Comment:
Is it possible for both GDPR and CCPA to apply? If so, does your endpoint support multiple `reg` values?

Reply:
Is it a requirement that we have to support both GDPR and CCPA when they both applies? We think it'd be unlikely that a user would be classified as under both GDPR and CCPA at the same time, and we currently only consider the GDPR consent if that's the case.

Reply:
I checked with other reviewers and it's ok to only consider GDPR in that case.

Reply:
every user in California browsing a website in the EU has both apply. To test this, set your IP to California and go to Guardian.co.uk and see how it considers this.

Reply:
As another datapoint in support of both frequently applying. Some publishers choose for GDPR to always apply, as considered in this document. https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md